### DISCUSSION TITLE
setIBLRotation:: how to get the rotation matrix in vertex block

### DISCUSSION THREAD

**Discussion by JackKaiXing** [google/filament#3957] [General]
Hello @romainguy ,

This discussion origins from issue **https://github.com/google/filament/issues/3951**.

Let me describe  what we are trying to achieve.

----------------------------------------

VBO 
**position attribute** in world space
`{0.0, 0.0, 0.0},{0.0, 0.0, 0.0},{1.0, 0.0, 0.0},{1.0, 0.0, 0.0}`
 **custom attribute** in world space, 
`{0.0, -1.0, 0.0},{0.0, 1.0, 0.0},{0.0, -1.0, 0.0},{0.0, 1.0, 0.0}`;

IBO
 `{0,2,3, 0,3,1}`;

Material Parameter
`width` : float type

vertex block

`vec4 pos = material.worldPosition; `
`vec2  dir = getCustom0().xy; `
`material.worldPosition = vec4(pos.x + dir.x * width, pox.y + dir.y * width, pos.z, pos.w);`

----------------------------------------

It works fine without setIBLRotation. What could help if I setIBLRotation?

Any other solution besides passing rotation matrix to vertex block?

**Comment by romainguy** [comment]
I'm still not sure what you are trying to do. You could pass the ibl rotation as a parameter indeed, it's a single value since the rotation is constrained to the Y axis.

Are you trying to keep a rectangle aligned to the camera? If so you may want to simply use a different vertex domain so you wouldn't have to deal with the IBL rotation at all.

**Reply by JackKaiXing** [reply]
1. what we are trying to do is to draw a Polyline with varying width at runtime.

   as the vbo, ibo listed above, if `width= 2`
   the final vertex world position is `{0.0, -2.0, 0.0},{0.0, 2.0, 0.0},{1.0, -2.0, 0.0},{1.0, 2.0, 0.0}`

   as the vbo, ibo listed above, if `width= 3`
   the final vertex world position is `{0.0, -3.0, 0.0},{0.0, 3.0, 0.0},{1.0, -3.0, 0.0},{1.0, 3.0, 0.0}`

2. setIBLRotation uses an axis and radians, is it a constrained to the Y axis?
